### PR TITLE
TJ Product Detail 

### PR DESCRIPTION
**Pull request by EliOJackson**
# Description

Set up our Product Detail controller to require in our Product Detail model and connect to the Product Detail route. When the route for Product Details is selected, the Details of the Product are listed in the dom.

Fixes Part of # 5. 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

To test, run 

```
 npm start
```
to begin running your sever.  Visit localhost8080 to pull up the log in screen. If you have not registered your account, select the register account button. Else, select the log in button and proceed to log in.

Once you have logged in, in your url enter the following address, replacing id with the id of the product you wish to view (1 - 11 please):

```
http://localhost:8080/product/<id>
```

You will then see the product details for the product you have selected!

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ x My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ router.get('/', (req, res, next) => {
 
 // pipe all other requests through the route modules
 router.use(require('./authRoute'));
+router.use(require('./productDetailRoute'));
```
Comment (Rian501) [inline routes/index.js]: As you go forward, you might want to be more inclusive about this, but it's not absolutely necessary. But won't productDetailRoute only have one route in it..?

Comment (Rian501) [reply routes/index.js]: ....nevermind. It probably will have edit/delete options too. Still, you may end up wanting to group your product routes together. This is up to you, however.

File: views/productDetail.pug [hunk 2]
```diff
@@ -0,0 +1,15 @@
+doctype html
+html
+  head
+    title Product Details
+    include stylesheets.pug
+  body
+    include ./partials/nav.pug
+    
+  block content
+    h1 Product Details
```
Comment (Rian501) [inline views/productDetail.pug]: You may want to provide for 'what if' a user inputs a url with a number or something the system can't handle. Just something to think about ....

File: controllers/productDetailCtrl.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+"use strict";
+
+const express = require("express");
+const app = express();
+const passport = require("passport");
+
+app.set("models", require("../models"));
```
Comment (Rian501) [inline controllers/productDetailCtrl.js]: the app.set should only really happen in the app.js, so you can require it through the req after that

File: routes/index.js (hunk 1 above)
Comment (EliOJackson) [reply routes/index.js]: Hey @Rian501, thanks for the great suggestion! However, we have decided to stick with our Product Detail Ctrl at this time. Let me know if you have any questions!

File: views/productDetail.pug (hunk 2 above)
Comment (EliOJackson) [reply views/productDetail.pug]: Hey @Rian501, thanks for the great suggestion! This is going to be handled by another teammate later on in the sprint. Let me know if you have any questions!

File: controllers/productDetailCtrl.js (hunk 3 above)
Comment (EliOJackson) [reply controllers/productDetailCtrl.js]: Hey @Rian501, thanks for the great suggestion! Per your request, I have updated our CTRL to correctly use req.app to achieve the same effect. Let me know if you have any questions!

**Pull request merged**